test/pgn_test.dart

```diff
@@ -276,6 +276,27 @@ the players are also trying to learn as much as possible about the opponent's pr
       expect(games.length, 3);
     });
 
+    test('Parse initial game comments - comment before newline', () {
+      final String data =
+          File('./data/wcc_2023_eolcomment.pgn').readAsStringSync();
+      final game = PgnGame.parsePgn(data);
+      expect(
+        game.comments,
+        [
```

Comment:
Not sure to understand what wasn't working before here. Parsing the `wcc_2023` was working fine as far as I can tell.

Sorry if I didn't get right your explanation in the PR description. Can you try to explain quickly what kind of bug is tested here? (perhaps as a code comment in the test so future readers will know too).

Thanks!

Reply:
You can find an explanation in the second part of the reply here: https://github.com/lichess-org/dartchess/pull/42#issuecomment-2398226884 I have also updated the test cases to include a description of the exact scenario being tested (#38).

Just to be clear and avoid any confusion, this PR is fixing both #30 and #38. Both issues are bugs in the PGN parser but in separate areas of the code.